app/models/shop.rb

```diff
@@ -12,12 +12,12 @@ def api_version
   end
 
   def self.store(auth_session)
-    ShopifyAPI::Base.activate_session(auth_session)
     remote_shop = ShopifyAPI::Shop.current
-    shop = Shop.find_or_initialize_by(shopify_domain: remote_shop.domain)
     customers = ShopifyAPI::Customer.all
     products = ShopifyAPI::Product.all
 
+    shop = Shop.find_or_initialize_by(shopify_domain: remote_shop.domain)
+
     shop.update_attributes(
```

Comment:
Please make separate methods for saving shop, customers and products.

Reply:
Also define attributes separately and call update separately.  e.g Shop.update!(shop_attributes) or
Shop.update!  shop_attributes

Reply:
Please look into "upsert" and see if we can use it here.

Reply:
Also look into iterators and how we can divide this work i.e. saving shop and customers details etc into different segments.